File: Interview Preparation Kit - CPP/02. Arrays/003. New Year Chaos.cpp
// Problem: https://www.hackerrank.com/challenges/new-year-chaos/problem
//  Score: 40

#include <iostream>
#include <vector>
using namespace std;


bool check_sorted(vector<int> & arr){
    bool ans = true;
    for (int i = 0; i < arr.size() - 1; i++){
        if (arr[i] > arr[i + 1]){
            ans = false;
        }
    }
    return ans;
}


int main(){
    int t;
    cin >> t;

    for (int i = 0; i < t; i++){
        int n;
        cin >> n;
        vector<int> arr;

        for (int j = 0; j < n; j++){
            int tmp;
            cin >> tmp;
            arr.push_back(tmp);
        }

        int ans = 0;

        for (int j = 0; j < 2; j++){
            for (int k = arr.size() - 1; k > 0; k--){
                if (arr[k] < arr[k - 1]){
                    swap(arr[k], arr[k - 1]);
                    ans++;
                }
            }
        }

        if (check_sorted(arr)){
            cout << ans << endl;
        }
        else {
            cout << "Too chaotic" << endl;
        }

    }

    return 0;
}

File: Interview Preparation Kit - Python/02. Arrays/003. New Year Chaos.py
# Problem: https://www.hackerrank.com/challenges/new-year-chaos/problem
#  Score: 40


t = int(input())
for test in range(t):
    n = int(input())
    arr = list(map(int, input().split()))
    count = 0

    for i in range(2):
        for j in range(len(arr) - 1, 0, -1):
            if arr[j] < arr[j-1]:
                arr[j], arr[j-1] = arr[j-1], arr[j]
                count += 1

    if arr == sorted(arr):
        print(count)
    else:
        print('Too chaotic')
